game_modules/game_engine.py

```diff
@@ -0,0 +1,822 @@
+    """
+    def duel_select(self):
+        pygame.init()
+
+        self.menuWidth = 800
+        self.menuHeight = 600
+        self.menuResolution = (self.menuWidth, self.menuHeight)
+        self.menu_screen = pygame.display.set_mode(self.menuResolution)
+        level = True
+        while level:
+            for event in pygame.event.get():
+                if event.type == QUIT:
+                    pygame.quit()
+                    quit()
+            self.menu_screen.fill(self.black)
+            self.button("HELPPO", 160, 200, 500, 75, self.green, self.light_green, "easy_duel")
+            self.button("NORMAALI", 160, 300, 500, 75, self.yellow, self.light_yellow, "normal_duel")
+            self.button("VAIKEA", 160, 400, 500, 75, self.red, self.light_red, "hard_duel")
+            # Näppäin ohjeet
+            # Pelaaja 1
+            self.menu_screen.blit(self.arrow_keys, (100+75, 75))
+            font = pygame.font.Font('fonts/OpenSans-Regular.ttf', 25)
+            self.drawText("Pelaaja 1:", font, self.white, (155+75, 150))
+            # Pelaaja 2
+            self.menu_screen.blit(self.wasd_keys, (225+85, 75))
+            font = pygame.font.Font('fonts/OpenSans-Regular.ttf', 25)
+            self.drawText("Pelaaja 2:", font, self.white, (280+85, 150))
+            # Äänet
+            self.menu_screen.blit(self.volume, (550+60, 90))
+            self.menu_screen.blit(self.volume_mute, (500+60, 92))
+            font = pygame.font.Font('fonts/OpenSans-Regular.ttf', 25)
+            self.drawText("M", font, self.white, (550+60, 150))
+            # Pause
+            self.menu_screen.blit(self.esc_key, (400+75, 90))
+            font = pygame.font.Font('fonts/OpenSans-Regular.ttf', 25)
+            self.drawText("Pysäytys", font, self.white, (418+75, 150))
+
+            pygame.display.update()
+            self.clock.tick(15)
+
+
+#################################
+# GAME LUOKKA
+#################################
+
+"""
+    Game -luokka, jonka avulla luomme itse peliympäristön.
+
+    Attributes
+    =========
+
+    windowWidth : int
+    windowHeight : int
+    screenResolution : tuple
+    clock : Clock
+    snake : Snake
+    other_snake : Snake
+    score1 : int
+    score2 : int
+
+    volume : Surface
+    volume_mute : Surface
+    arrow_keys : Surface
+    wasd_keys : Surface
+    
+    mouse : tuple
+    click : boolean
+    
+    bite_sound : Sound
+    fail_sound : Sound
+    click_sound : Sound
+    
+    running : boolean
+    last : int
+    last2 : int
+    cooldown : int
+    difficulty : Difficulty(enum)
+    gamemode : Gamemode(enum)
+    obstacles : Array
+    pause : boolean
+    game_over : boolean
+    sound : boolean
+
+    Params
+    ======
+
+    difficulty : Difficulty(enum)
+    gamemode : Gamemode(enum)
+
+    Methods
+    =======
+
+    gameLoop : void
+    startGame : void
+    drawGrid : void
+    drawText : void
+    text_object : Surface, Rect
+
+"""
+
+class Game:
```

Comment:
Game luokasta voisi ottaa irti funktionaalisuutta muihin luokkiin? Esim ääni ja ruudun piirtäminen. SRP ja hard to reuse.